server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -252,7 +252,14 @@ private HitContext prepareNonNestedHitContext(SearchContext context,
         int subDocId = docId - subReaderContext.docBase;
         if (fieldsVisitor == null) {
             SearchHit hit = new SearchHit(docId, null, null, null);
-            return new HitContext(hit, subReaderContext, subDocId, context.searcher(), sharedCache);
+            return new HitContext(
+                hit,
+                subReaderContext,
+                subDocId,
+                context.searcher(),
+                context.getQueryShardContext().fetchLookup().source(),
```

Comment:
Now that all of the phases get a `SearchLookup` when they are built we *could* remove this and rely on them closing over what they need at construction time. That is pretty much what the doc values fetcher does, but not all of them do it. I think it is a bit confusing that some do and some don't, but I don't think now is the right time to make that change.